app/script/view_model/content/InputBarViewModel.js

```diff
@@ -101,12 +101,10 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
         .reverse()
         .reduce(
           (currentPieces, mentionEntity) => {
-            const {end: mentionEnd, start: mentionStart} = mentionEntity;
-
             const currentPiece = currentPieces.shift();
-            currentPieces.unshift(currentPiece.substr(mentionEnd));
-            currentPieces.unshift(currentPiece.substr(mentionStart, mentionEntity.getLength()));
-            currentPieces.unshift(currentPiece.substr(0, mentionStart));
+            currentPieces.unshift(currentPiece.substr(mentionEntity.endIndex + 1));
```

Comment:
After the change to `MentionEntity` => `currentPiece.substr(mentionEntity.endIndex)`